testRearrangeCarGroup8.py

```diff
@@ -0,0 +1,118 @@
+import unittest
+import numpy as np
+from RearrangeCarGroup8 import RearrangeCars
+
+class TestRearrangeCars(unittest.TestCase):
+
+    '''Unittests for RearrangeCars class and promised fuctionality
+    '''
+
+    def setUp(self):
+
+        '''Basic setting that will be used in several tests
+        '''
+        
+        self.initial_car_order = [1, 2, 0, 3]
+        self.final_car_order = [3, 1, 2, 0]
+        self.rc = RearrangeCars(list(self.initial_car_order), list(self.final_car_order))
+
+    def test_empty_parking_lot(self):
+
+        '''Test the case when the parking lot is empty,
+        should work without raising an error
+        '''
+        
+        rc_empty = RearrangeCars(list(), list())

```

Comment:
And here it is better to pass just two "[]" parameters as list creation with "[", "]" is significantly faster than by list() and also it will be clearer what order we have and what are willing to obtain.